### PR TITLE
Add ADOPTERS.md for user list

### PR DESCRIPTION
**Pull request by rothgar**
**1. Description of changes:**
Adds a file so karpenter users can self report

**2. How was this change tested?**
it wasn't

**17. Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ellistarn — COMMENTED** [review]
Great idea!

Thoughts on calling this ADOPTERS.md?
https://github.com/knative/community/blob/main/ADOPTERS.MD

File: USERS.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Who is using Karpenter?
+Karpenter has a variety of users and use cases for scaling Kubernetes.
+Many customers want to learn from others who have already implemented Karpenter in their environments.
+
+The following is a self-reported list of users to help identify adoption and points of contact.
+
+## Community
+If you would like to ask question from the community please join the [Karpenter slack channel in the Kubernetes Slack](https://kubernetes.slack.com/archives/C02SFFZSA2K) or join the [Karpenter working group](WORKING_GROUP.md) and fort-nightly calls.
```
Comment (njtran) [inline USERS.md:8]: Definitely great use of fortnight here, but I think we should stick to bi-weekly so someone doesn't have to look it up. 
```suggestion
If you would like to ask question from the community please join the [Karpenter slack channel in the Kubernetes Slack](https://kubernetes.slack.com/archives/C02SFFZSA2K) or join the [Karpenter working group](WORKING_GROUP.md) bi-weekly calls.
```

**Review by njtran — COMMENTED** [review]
Love the idea. Thanks!

File: USERS.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+* `@Alex Kestner`, `@Ellis Tarn`
+
+```
+* *Company*
+* Description of where Karpenter is used
+* _Link to company or blog/details_ (optional)
+* `slack user`, `email@company` (optional)
+```
+
+## Users (Alphabetically)
```
Comment (njtran) [inline USERS.md:28]: Hard to see in the preview if this works, but you can try this.
```suggestion
## Adopters

| Organization                      | Product/Project Name                                                   | Status                   | More Information                                                     |                K8s Slack Contact
| --------------------------------- | ---------------------------------------------------------------------- | ------------------------ | -------------------------------------------------------------------- |
| Amazon, Inc.                      | Use-Case                                               | Production               | [link](https://cs.console.aliyun.com/)                               |            `@Alex Kestner` and `@Ellis Tarn`
```

Comment (ellistarn) [reply USERS.md:28]: This redirects me to alibaba: https://cs.console.aliyun.com/

**Comment by rothgar** [comment]
> Great idea!
> 
> Thoughts on calling this ADOPTERS.md? https://github.com/knative/community/blob/main/ADOPTERS.MD

I like that table format better than what I have. I based this PR on cilium's users https://github.com/cilium/cilium/blob/master/USERS.md

**Comment by ellistarn** [comment]
> Does anyone have strong opinions on calling the file USERS or ADOPTERS? I see a mix of both on github

Mild preference against "USERS" because of the drug connotation 🤷 .

**Review by ellistarn — APPROVED** [review]

File: USERS.md (hunk 2 above)
Comment (njtran) [reply USERS.md]: oops!

**Pull request merged**